src/main/kotlin/com/exactpro/th2/common/schema/message/ConfirmationMessageListener.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2020-2022 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.th2.common.schema.message
+
+interface ConfirmationMessageListener<T> {
+    /**
+     * If returns `true` that means the listener handles confirmation by himself
+     * and no auto-confirmation should be performed.
+     *
+     * If returns `false` the confirmation should be performed automatically when method [handle] returns
+     */
+    val manualConfirmation: Boolean
+
+    @Throws(Exception::class)
+    fun handle(consumerTag: String, message: T, confirmation: ManualAckDeliveryCallback.Confirmation)
+
+    fun onClose() {}
+
+    companion object {
+        @JvmStatic
+        @JvmOverloads
+        fun <T> wrap(listener: MessageListener<T>, confirm: Boolean = false): ConfirmationMessageListener<T> = DelegateListener(listener, confirm)
```

Comment:
Maybe the confirm option is always true because we move the autoconfirm logic for the ConnectionManager and the listener don't know about ManualAckDeliveryCallback.Confirmation

Reply:
In case there is more than one listener for one queue and one of them has manual acknowledgment the other ones should not confirm the delivery.  That is why the default value for `confirm` parameter is `false`

Reply:
But maybe we should use `confirm` parameter as a value to return from `manualConfirmation` property

Reply:
I didn't find a pace where the wrap method is called with the confirm argument